src/org/joni/ByteCodeMachine.java

```diff
@@ -60,6 +60,9 @@
 
     public void interrupt() {
         interrupted = true;
+        // might have no effect on the executing thread but worth a try
+        // we might not succeed interrupting on next loop but will eventually
+        synchronized (this) { INTERRUPT_CHECK_EVERY = 0; }
```

Comment:
Having this all caps confused me into thinking it was updating a global constant.

Reply:
this started as a real constant, renamed - thanks for the :eyes: